autorest/azure/azure.go

```diff
@@ -285,26 +286,45 @@ func WithErrorUnlessStatusCode(codes ...int) autorest.RespondDecorator {
 				var e RequestError
 				defer resp.Body.Close()
 
+				encodedAs := autorest.EncodedAsJSON
+				if strings.Contains(resp.Header.Get("Content-Type"), "xml") {
+					encodedAs = autorest.EncodedAsXML
+				}
+
 				// Copy and replace the Body in case it does not contain an error object.
 				// This will leave the Body available to the caller.
-				b, decodeErr := autorest.CopyAndDecode(autorest.EncodedAsJSON, resp.Body, &e)
+				b, decodeErr := autorest.CopyAndDecode(encodedAs, resp.Body, &e)
 				resp.Body = ioutil.NopCloser(&b)
 				if decodeErr != nil {
 					return fmt.Errorf("autorest/azure: error response cannot be parsed: %q error: %v", b.String(), decodeErr)
 				}
 				if e.ServiceError == nil {
 					// Check if error is unwrapped ServiceError
-					if err := json.Unmarshal(b.Bytes(), &e.ServiceError); err != nil {
-						return err
+					if encodedAs == autorest.EncodedAsXML {
+						if err := xml.Unmarshal(b.Bytes(), &e.ServiceError); err != nil {
+							return err
+						}
+					} else {
+						if err := json.Unmarshal(b.Bytes(), &e.ServiceError); err != nil {
+							return err
+						}
 					}
 				}
 				if e.ServiceError.Message == "" {
 					// if we're here it means the returned error wasn't OData v4 compliant.
-					// try to unmarshal the body as raw JSON in hopes of getting something.
+					// try to unmarshal the body in hopes of getting something.
 					rawBody := map[string]interface{}{}
-					if err := json.Unmarshal(b.Bytes(), &rawBody); err != nil {
-						return err
+
+					if encodedAs == autorest.EncodedAsXML {
+						if err := xml.Unmarshal(b.Bytes(), &rawBody); err != nil {
```

Comment:
Does this actually work for XML?

Reply:
as far as I'm aware - there's some examples [in this post](https://github.com/Azure/go-autorest/pull/465#issue-310460477) but I can add some tests too?

Reply:
Adding a test for XML would be great.